databricks/koalas/tests/test_indexes.py

```diff
@@ -1001,3 +1001,147 @@ def test_unique(self):
 
         self.assert_eq(kmidx.unique().sort_values(), pmidx.unique().sort_values())
         self.assert_eq(kmidx.unique().sort_values(), pmidx.unique().sort_values())
+
+    def test_union(self):
+        # Index
+        pidx1 = pd.Index([1, 2, 3, 4])
+        pidx2 = pd.Index([3, 4, 5, 6])
+        kidx1 = ks.from_pandas(pidx1)
+        kidx2 = ks.from_pandas(pidx2)
+
+        self.assert_eq(kidx1.union(kidx2), pidx1.union(pidx2))
+        self.assert_eq(kidx2.union(kidx1), pidx2.union(pidx1))
+        self.assert_eq(
+            kidx1.union([3, 4, 5, 6]), pidx1.union([3, 4, 5, 6]),
+        )
+        self.assert_eq(
+            kidx2.union([1, 2, 3, 4]), pidx2.union([1, 2, 3, 4]),
+        )
```

Comment:
Shall we add one more test using `Series`?
Seems like we don't have the test for `pandas<0.24`.

Reply:
Sure, just added it.
Thanks for finding and also other comments above :D